cmd/jujud/agent/machine.go

```diff
@@ -985,6 +986,17 @@ func (a *MachineAgent) startStateWorkers(st *state.State) (worker.Worker, error)
 			a.startWorkerAfterUpgrade(runner, "mongoupgrade", func() (worker.Worker, error) {
 				return newUpgradeMongoWorker(st, a.machineId, a.maybeStopMongo)
 			})
+			a.startWorkerAfterUpgrade(runner, "statemetrics", func() (worker.Worker, error) {
+				return worker.NewSimpleWorker(func(stop <-chan struct{}) error {
```

Comment:
I've split out the code to a separate function. The agent is kind of a mess, testing it in isolation is currently impractical